compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11995,15 +11995,27 @@ OMR::Z::TreeEvaluator::lRegLoadEvaluator(TR::Node * node, TR::CodeGenerator * cg
 TR::Register *
 OMR::Z::TreeEvaluator::iRegStoreEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
-   TR::Node * child = node->getFirstChild();
+   TR::Node* value = node->getFirstChild();
+
+   TR::Register* globalReg = NULL;
+
+   // If the child is an add/sub that could be used in a branch on count op, don't evaluate it here
```

Comment:
Can you please clarify what these lines changed have to do with the problem described in the commit message ? This seems to be avoiding evaluating the RHS value of a reg store altogether whereas the bug that was being talked about was around some value child not doing some extension that the parent was assuming would have happened.

Reply:
This sequence is preset in the current evaluator as well. See [1]. The reason you see the change in green in the diff is because I moved it up to the top of the evaluator since the fast-path exit in this code does not rely on any other parts of the same evaluator and hence can be done first.

[1] https://github.com/eclipse/omr/blob/deaaf6ce60e5878101de33efbfe14054bedff025/compiler/z/codegen/OMRTreeEvaluator.cpp#L12029-L12038

Reply:
Okay I think how this is related. The code later was omitting evaluating some child in some cases but was setting a flag that was causing extension to be skipped. Now we omit evaluating the child early such that the flag also won't be set.

Reply:
Once I get confirmation that I've understood this right, I'll merge.

Reply:
Your understanding is spot on.